symengine/as_real_imag.cpp

```diff
@@ -0,0 +1,284 @@
+#include <symengine/visitor.h>
+#include <symengine/basic.h>
+
+namespace SymEngine
+{
+
+class RealImagVisitor : public BaseVisitor<RealImagVisitor>
+{
+private:
+    Ptr<RCP<const Basic>> real_, imag_;
```

Comment:
I think we should not use `Ptr` here. Just `RCP<const Basic> real_, imag_;` and have a method to extract `real_` and `imag_` using `Ptr`s